treewide: move fixDarwinDylibNames to nativeBuildInputs

thefloweringash opened the pull request:
###### Motivation for this change

Hopefully simple cleanup. I discovered some inconsistencies when cross compiling where install_name_tool has a target prefix, and checked through the uses in the entire tree.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

thefloweringash committed:
treewide: move fixDarwinDylibNames to nativeBuildInputs

This hook runs at build time and depends on executing
install_name_tool from binutils.

LnL7 commented on pkgs/tools/security/sudolikeaboss/default.nix:
```diff
@@ -19,9 +19,11 @@ buildGoPackage rec {
-    fixDarwinDylibNames
   ];
 
+  propagatedNativeBuildInputs =
```
This should never be propagated, that makes it impossible to build libraries rpath or executable relative references when this is a dependency.

LnL7 reviewed (changes requested):
Looks good to me otherwise.

thefloweringash committed:
sudolikeaboss: remove input propagation and unused hook

This package is a leaf that exports no libraries, so propagating its
inputs should have no effect.

The fixDarwinDylibNames hook only applies to *.dylib files, which this
package does not contain.

FRidh commented:
needs a rebase

thefloweringash commented:
Rebased!
